feat(COC): add COC scheduled task resource (#7750)
huaweicloud/terraform-provider-huaweicloud: docs/resources/coc_scheduled_task.md

* `associated_task_name` - (Required, String) Specifies the name of the associated task (script name/job name).

* `risk_level` - (Required, String) Specifies the risk level of the scheduled task. The default value is **HIGH**.
  Values can be as follows:
  + **HIGH**: High risk.
  + **MEDIUM**: Medium risk.
  + **LOW**: Low risk.

* `input_param` - (Required, Map) Specifies the task execution parameters.

* `target_instances` - (Required, List) Specifies the target instance information.

  The [target_instances](#target_instances_struct) structure is documented below.

  -> **NOTE:** If the task is associated with a script, this array has only one element, which specifies the resource
  instance that the script operates on.
  <br>If the task is associated with a job and the target instance is **SAME**, this array has only one element, which
  specifies the resource instance that the job operates on.
  <br>If the task is associated with a job and the target instance is **DIFF**, the number of elements in this array is
  equal to the number of steps in the associated job. That is, each task under each step specifies the resource instance
  to operate on.

> * `enable_approve` - (Required, Bool) Specifies whether to enable manual review of incoming shipments.

Reviewer: ```suggestion
* `enable_approve` - (Required, Bool) Specifies whether to enable manual review.
```

* `enable_message_notification` - (Required, Bool) Specifies whether to enable message notifications.

> * `ticket_infos` - (Optional, List) Specifies the order No. 4 rights request information.

Reviewer: Is the description correct?

* `policy` - (Required, String) Specifies the scheduled task execution strategy. The default value is **ONCE**.
  Values can be as follows:
  + **PERIODIC**: Periodic execution.
  + **ONCE**: Single execution.
  + **CRON**: Execute according to a CRON expression.

* `single_scheduled_time` - (Optional, Int) Specifies the execution time of a single scheduled task.

  -> This value is required if the scheduled task execution policy is **ONCE**. UTC timestamp in milliseconds.

* `periodic_scheduled_time` - (Optional, String) Specifies the daily execution time for periodic scheduled tasks.

  -> This value is required if the scheduled task execution policy is **PERIODIC**. A 24-hour time string. For example,
  if a task is to be executed at 5:30 PM on a given day, it would be **17:30:00**.

* `period` - (Optional, String) Specifies the specific week list for periodic scheduled tasks.

  -> This value is required if the scheduled task execution policy is **PERIODIC**. Days of the week are separated by
  commas. For example, Sunday is **1** and Monday is **3**. If a task is to be executed on Monday, Wednesday, Thursday,

>   and Sunday, the value would be **1,2,4,5**. The default value is *.

Reviewer: The default value is ？

* `cron` - (Optional, String) Specifies the specific CRON expression value for executing a scheduled task based on a
  CRON expression.

  -> This value is required if the scheduled task execution policy is **CRON**. A valid CRON expression is sufficient.
  For example, to execute a task at 10:15 AM every day, use ```15 10 ? * *```.

* `scheduled_close_time` - (Optional, Int) Specifies the deadline for scheduled task execution.

  -> This value is required if the scheduled task execution policy is **PERIODIC** or **CRON**. It specifies the
  timestamp of the scheduled task rule deadline. UTC timestamp in milliseconds.

<a name="target_instances_struct"></a>
The `target_instances` block supports:

* `target_selection` - (Required, String) Specifies the target instance selection method. The default value is **MANUAL**.
  Values can be as follows:
  + **ALL**: All instances.
  + **MANUAL**: Manual selection.
  + **NONE**: No instance specified.

* `order_no` - (Required, Int) Specifies the step number.

  -> **NOTE:** When a scheduled task is associated with a script, this value is **1**.

>   <br>When a scheduled task is associated with a job, this value indicates the step number within the job.

Reviewer: ```suggestion
* `order_no` - (Required, Int) Specifies the step number.
  + When a scheduled task is associated with a script, this value is **1**.
  + When a scheduled task is associated with a job, this value indicates the step number within the job.
```

  -> This field is required if the `target_selection` is **MANUAL**.

* `batch_strategy` - (Optional, String) Specifies the instance batching strategy.
  Values can be as follows:
  + **AUTO_BATCH**: Automatic batching.
  + **MANUAL_BATCH**: Manual batching.
  + **NONE**: No batching.

  -> When a scheduled task is associated with a job and the current step does not require a resource instance, this
  value must be **NONE**.
  <br>When the `target_selection` is **ALL**, this value must be **AUTO_BATCH**.
  <br>When the resource instance list is not empty, this value can only be **AUTO_BATCH** or **MANUAL_BATCH**.

* `sub_target_instances` - (Optional, List) Specifies the secondary resource instance information.

  The [sub_target_instances](#target_instances_struct) structure is documented below.

  -> If a step in a job associated with a scheduled task has multiple tasks and the `runbook_instance_mode` is **DIFF**,
  this value is required. This means that each task specifies a separate resource instance for the operation.

<a name="target_instances_target_resource_struct"></a>
The `target_resource` block supports:

> * `type` - (Optional, String) Specifies the resource selection method (REGION, APPLICATION). The default value is **REGION**.

Reviewer: ```suggestion
* `type` - (Optional, String) Specifies the resource selection method. The default value is **REGION**.
```

  Values can be as follows:
  + **REGION**: Selects resource instances by region.
  + **APPLICATION**: Selects resource instances by application.

* `id` - (Optional, String) Specifies the region or application ID.

> * `app_name` - (Optional, String) Specifies the application name (separate hierarchical elements with .).

Reviewer: ```suggestion
* `app_name` - (Optional, String) Specifies the application name. Separate hierarchical elements with dot(.).
```

  -> When the resource selection method is **APPLICATION**, this value is required and specifies the name of the
  selected application.

* `region_id` - (Optional, String) Specifies the region ID which the application is associated.

  -> When the resource selection method is **APPLICATION**, this value is required.

* `params` - (Optional, List) Specifies the dynamic query conditions for resource instances.

  The [params](#target_instances_target_resource_params_struct) structure is documented below.

<a name="target_instances_target_resource_params_struct"></a>
The `params` block supports:

* `key` - (Required, String) Specifies the resource attribute name. For example, **ep_id**, **agent_state** and so on.

* `value` - (Required, String) Specifies the parameter value corresponding to the resource attribute.

<a name="ticket_infos_struct"></a>
The `ticket_infos` block supports:

* `ticket_id` - (Optional, String) Specifies the ID of order No. 4.

* `ticket_type` - (Optional, String) Specifies the type of order No. 4.
  Values can be as follows:
  + **CHANGE**: Change order.
  + **INCIDENT**: Incident order.

>   + **ALARM**: Alarm order.

Reviewer: ALARM is not at the API docs? Is it correct?

  + **WARROOM**: War room order.

> * `target_id` - (Optional, String) Specifies the application ID associated with the order No. 4.

Reviewer: add a line before this

* `scope_id` - (Optional, String) Specifies the region ID associated with the order No. 4.

<a name="message_notification_struct"></a>
The `reviewer_notification` and `message_notification` block supports:

* `notification_endpoint_type` - (Required, String) Specifies the notification policy.
  Values can be as follows:
  + **START_EXECUTION**: Starts execution.
  + **EXECUTION_FAILED**: Execution failed.
  + **EXECUTION_SUCCEEDED**: Execution succeeded.

  -> This value is required when `enable_message_notification` is **true**.

* `policy` - (Optional, String) Specifies the notification recipient type. The default value is **ONCALL**.
  Values can be as follows:
  + **USER**: Individual.

>   + **ONCALL**: Scheduled.

Reviewer: the description is mismatch, should swap each other

* `single_scheduled_time` - (Optional, Int) Specifies the execution time of a single scheduled task.

  -> This value is required if the scheduled task execution policy is **ONCE**. UTC timestamp in milliseconds.

* `periodic_scheduled_time` - (Optional, String) Specifies the daily execution time for periodic scheduled tasks.

  -> This value is required if the scheduled task execution policy is **PERIODIC**. A 24-hour time string. For example,
  if a task is to be executed at 5:30 PM on a given day, it would be **17:30:00**.

* `period` - (Optional, String) Specifies the specific week list for periodic scheduled tasks.

  -> This value is required if the scheduled task execution policy is **PERIODIC**. Days of the week are separated by
  commas. For example, Sunday is **1** and Monday is **3**. If a task is to be executed on Monday, Wednesday, Thursday,
  and Sunday, the value would be **1,2,4,5**.

* `cron` - (Optional, String) Specifies the specific CRON expression value for executing a scheduled task based on a
  CRON expression.

  -> This value is required if the scheduled task execution policy is **CRON**. A valid CRON expression is sufficient.

>   For example, to execute a task at 10:15 AM every day, use ```0 15 10 ? * *```.

Reviewer: ```suggestion
  For example, to execute a task at 10:15 AM every day, use `0 15 10 ? * *`.
```
